### PR TITLE
Stable-2.9.3

### LINKED ISSUE #5272: Updated TLS secrets not being picked up by tap or proxy-injector
### What is the issue?

We recently upgraded from `edge 20.2.2` to `stable 2.9.0` using the official Helm chart. We followed the setup documented [here](https://linkerd.io/2/tasks/automatically-rotating-webhook-tls-credentials/) in order to automatically rotate our webhook TLS credentials with cert manager. These certificates are being rotated correctly and the associated TLS secrets updated.

The issue is that after the TLS secrets are rotated by cert-manager, the `tap` and `proxy-injector` pods start producing the following error:
```
http: TLS handshake error from 127.0.0.1:52008: remote error: tls: bad certificate
```
These pods have to be manually restarted at which point they pick up the secrets correctly and are fine until the next certificate refresh.

We are using ArgoCD to manage deployments and Cert manager is version `0.15.1`.

#### `linkerd check` output

```text
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ control plane PodSecurityPolicies exist

linkerd-identity
----------------
√ certificate config is valid
√ trust anchors are using supported crypto algorithm
√ trust anchors are within their validity period
√ trust anchors are valid for at least 60 days
√ issuer cert is using supported crypto algorithm
√ issuer cert is within its validity period
‼ issuer cert is valid for at least 60 days
    issuer certificate will expire on 2020-11-23T20:07:30Z
    see https://linkerd.io/checks/#l5d-identity-issuer-cert-not-expiring-soon for hints
√ issuer cert is issued by the trust anchor

linkerd-webhooks-and-apisvc-tls
-------------------------------
√ tap API server has valid cert
‼ tap API server cert is valid for at least 60 days
    certificate will expire on 2020-11-24T02:25:03Z
    see https://linkerd.io/checks/#l5d-webhook-cert-not-expiring-soon for hints
√ proxy-injector webhook has valid cert
‼ proxy-injector cert is valid for at least 60 days
    certificate will expire on 2020-11-24T02:25:03Z
    see https://linkerd.io/checks/#l5d-webhook-cert-not-expiring-soon for hints
√ sp-validator webhook has valid cert
‼ sp-validator cert is valid for at least 60 days
    certificate will expire on 2020-11-24T02:25:03Z
    see https://linkerd.io/checks/#l5d-webhook-cert-not-expiring-soon for hints

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ tap api service is running

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

control-plane-version
---------------------
√ control plane is up-to-date
√ control plane and cli versions match

linkerd-ha-checks
-----------------
√ pod injection disabled on kube-system

linkerd-prometheus
------------------
√ prometheus add-on service account exists
√ prometheus add-on config map exists
√ prometheus pod is running

linkerd-grafana
---------------
√ grafana add-on service account exists
√ grafana add-on config map exists
√ grafana pod is running

Status check results are √
```

### Environment

- Kubernetes Version: 1.16
- Cluster Environment: EKS
- Host OS: Amazon Linux
- Linkerd version: 2.9.0

### LINKED ISSUE #5298: proxy error:client requires absolute-form URIs
### What is the issue?

~15 hours after upgrading our staging cluster from edge-20.7.5 to stable-2.9.0, we observed connectivity errors between multiple pods in the same namespace that normally communicate over k8s services.  From the perspective of the client, all requests to `http://server/foo` were returned instantly with a 502 error, while the linkerd-proxy log on both the client and server reported e.g:

```
[  8109.623316s]  WARN ThreadId(01) inbound:accept{peer.addr=10.4.2.144:50078 target.addr=10.4.2.205:3000}: linkerd2_app_core::errors: Failed to proxy request: client requires absolute-form URIs
```

Note that the pods were _not_ restarted after the linkerd update, so while the control plane was running 2.9.0, the pods were still running the older version of the proxy.

Interestingly, if from the client pod I manually curl `http://server.namespace/foo` rather than `http://server/foo` the request succeeds; it's only the un-qualified version of the URL that fails.

### How can it be reproduced?

At least for us, the repro path is to update from the old edge release to the new stable release and then wait 12-24 hours; it's happened twice so far.  That said I do not know what level of traffic (if any) might be necessary to trigger it and there is no obvious reason why the issue starts when it does.

#### `linkerd check` output

Pre-update: 

```text
$ linkerd check
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version
√ is running the minimum kubectl version

linkerd-existence
-----------------
√ 'linkerd-config' config map exists
√ heartbeat ServiceAccount exist
√ control plane replica sets are ready
√ no unschedulable pods
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-config
--------------
√ control plane Namespace exists
√ control plane ClusterRoles exist
√ control plane ClusterRoleBindings exist
√ control plane ServiceAccounts exist
√ control plane CustomResourceDefinitions exist
√ control plane MutatingWebhookConfigurations exist
√ control plane ValidatingWebhookConfigurations exist
√ control plane PodSecurityPolicies exist

linkerd-identity
----------------
√ certificate config is valid
√ trust roots are using supported crypto algorithm
√ trust roots are within their validity period
√ trust roots are valid for at least 60 days
√ issuer cert is using supported crypto algorithm
√ issuer cert is within its validity period
√ issuer cert is valid for at least 60 days
√ issuer cert is issued by the trust root

linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ tap api service is running

linkerd-version
---------------
√ can determine the latest version
‼ cli is up-to-date
    is running version 2.7.1 but the latest stable version is 2.9.0
    see https://linkerd.io/checks/#l5d-version-cli for hints

control-plane-version
---------------------
‼ control plane is up-to-date
    is running version 20.7.5 but the latest edge version is 20.11.5
    see https://linkerd.io/checks/#l5d-version-control for hints
‼ control plane and cli versions match
    control plane running edge-20.7.5 but cli running stable-2.7.1
    see https://linkerd.io/checks/#l5d-version-control for hints

Status check results are √```

### Environment

- Kubernetes Version: 1.16.15-gke.4300
- Cluster Environment: GKE
- Host OS: Google Container-Optimized OS (cos)
- Linkerd version: edge-20.7.5; stable-2.9.0

### LINKED ISSUE #5595: Add 5432 to default skip ports?
We've had a bunch of issues recently referencing port 5432 which is used by Amazon RDS and, I believe Postgres. If we do a stable-2.9.3, should we include this skip port? We'll still need to fold this into https://github.com/linkerd/linkerd2/issues/5571 for stable-2.10...

### PR DESCRIPTION
**Pull request by adleong**
This stable release fixes an issue that prevented the proxy from being able
to speak HTTP/1 with older versioned proxies. It also fixes an issue where the
`linkerd-config-overrides` secret would be deleted during upgrade and provides
a `linkerd repair` command for restoring it if it has been deleted.

* Fixed an issue that could cause the inbound proxy to fail meshed HTTP/1
  requests from older proxies (from the stable-2.8.x vintage)
* Fixed an issue where the Linkerd webhooks and apiservices would not refresh 
  their certs automatically when provided externally — like through cert-manager
* Added missing label `linkerd.io/control-plane-ns` to the
 `linkerd-config-overrides` secret to prevent it from being pruned during
 upgrades
* Added `linkerd repair` command to restore the `linkerd-config-overrides`
  secret if it has been pruned
* Added port 5432 which is used by Amazon RDS and Postgres to the default list
  of skipped ports

### CODE REVIEW

**Comment by alpeb** [comment]
For some reason using external certs for the injector isn't working for me, to be able to test the webhook secret reloading. I'll give it another try on Monday.

**Comment by alpeb** [comment]
I finally tested successfully #5272 (the problem  was I was manually generating certs with no SAN) :+1: I tested specifically  refreshing the tap and injector certs.

**Comment by adleong** [comment]
I have tested upgrading from stable-2.8.1 to this version (with the --prune flag) and verified that `secret/linkerd-config-overrides` exists after the upgrade and that it has the correct label.

**Comment by adleong** [comment]
I have tested cross-version HTTP communication by doing the following:

* install Linkerd stable-2.8.1
* inject and install emojivoto
* upgrade Linkerd to this version
* restart vote-bot to upgrade the vote-bot proxy to this version
  * (vote-bot is now running this version and web is running stable-2.8.1) 
* verified vote-bot outbound success rate is still good (other than intended application errors)
* exec into vote-bot container and manually verify that I can curl the web service successfully

@olix0r is this testing sufficient to exercise #5298?

**Comment by adleong** [comment]
Verified that pods injected with this version have 5432 in the init container skip argument:

```
  initContainers:
  - args:
    - --incoming-proxy-port
    - "4143"
    - --outgoing-proxy-port
    - "4140"
    - --proxy-uid
    - "2102"
    - --inbound-ports-to-ignore
    - 4190,4191,25,443,587,3306,11211,5432
    - --outbound-ports-to-ignore
    - 25,443,587,3306,11211,5432
 ```

**Comment by adleong** [comment]
Verified the `linkerd repair` workflow:

* Install Linkerd stable-2.8.1
* Upgrade to Linkerd stable-2.9.2 using --prune
  * (secret/linkerd-config-overrides is deleted)
* Using this version of the CLI, verify that attempting to run `linkerd upgrade` prints a useful error message
* Using this version of the CLI, run `linkerd repair | kubectl apply -f -`
* Upgrade to this version of the control plane

**Review by dadjeibaah — APPROVED** [review]
LGTM 👍

File: charts/linkerd2/values.yaml [hunk 1]
```diff
@@ -95,8 +95,9 @@ global:
     # - HTTPS (443) opaque TLS
     # - MYSQL (3306) server-first
     # - Memcached (11211) clients do not issue any preamble, which breaks detection
-    ignoreInboundPorts: "25,443,587,3306,11211"
-    ignoreOutboundPorts: "25,443,587,3306,11211"
+    # - Postgres (5432) clients do not issue any preamble, which breaks detection
+    ignoreInboundPorts: "25,443,587,3306,11211,5432"
+    ignoreOutboundPorts: "25,443,587,3306,11211,5432"
```
Comment (olix0r) [inline charts/linkerd2/values.yaml:100]: Ideally these would be sorted

**Review by olix0r — APPROVED** [review]
I would be okay omitting the 5432 skip-ports change, as we haven't actually tested that on edges. But I think it's also basically harmless to include, so 🤷‍♂️ shipit.

**Review by Pothulapati — APPROVED** [review]
LGTM, Performed some basic demo testing along with upgrades with repair cmd. Repair correctly restores overridden configuration 👍

**Review by zaharidichev — APPROVED** [review]
LGTM

**Pull request merged**